packages/react-core/src/demos/examples/JumpLinks/JumpLinksWithDrawer.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import {
+  Button,
+  Drawer,
+  DrawerPanelContent,
+  DrawerContent,
+  DrawerContentBody,
+  DrawerHead,
+  DrawerActions,
+  DrawerCloseButton,
+  JumpLinks,
+  JumpLinksItem,
+  JumpLinksList,
+  Sidebar,
+  SidebarContent,
+  SidebarPanel,
+  TextContent,
+  getResizeObserver
+} from '@patternfly/react-core';
+import DashboardWrapper from '../DashboardWrapper';
+
+export const JumpLinksWithDrawer = () => {
+  const headings = ['First', 'Second', 'Third', 'Fourth', 'Fifth'];
+
+  const [offsetHeight, setOffsetHeight] = React.useState(0);
+  const [isExpanded, setIsExpanded] = React.useState(false);
+  const drawerRef = React.useRef();
+
+  React.useEffect(() => {
+    const masthead = document.getElementsByClassName('pf-c-masthead')[0];
+
+    getResizeObserver(masthead, () => {
+      setOffsetHeight(masthead.offsetHeight);
+    });
+  }, []);
+
+  const onCloseClick = () => {
+    setIsExpanded(false);
+  };
+
+  const onToggleClick = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  const panelContent = (
+    <DrawerPanelContent>
+      <DrawerHead>
+        <span tabIndex={isExpanded ? 0 : -1} ref={drawerRef}>
+          drawer-panel
+        </span>
+        <DrawerActions>
+          <DrawerCloseButton onClick={onCloseClick} />
+        </DrawerActions>
+      </DrawerHead>
+    </DrawerPanelContent>
+  );
+
+  return (
+    <DashboardWrapper breadcrumb={null} mainContainerId="scrollable-element">
+      <Drawer isExpanded={isExpanded}>
+        <DrawerContent panelContent={panelContent} id="jump-links-drawer-drawer-scrollable-container">
+          <DrawerContentBody>
+            <DrawerContentBody hasPadding>
+              <Sidebar>
+                <SidebarPanel variant="sticky">
+                  <JumpLinks
+                    isVertical={true}
+                    label="Jump to section"
+                    scrollableSelector="#jump-links-drawer-drawer-scrollable-container"
+                    offset={offsetHeight}
+                    expandable={{ default: 'expandable', md: 'nonExpandable' }}
+                  >
+                    {headings.map(heading => (
+                      <JumpLinksItem key={heading} href={`#jump-links-drawer-jump-links-${heading.toLowerCase()}`}>
+                        {`${heading} section`}
+                        <JumpLinksList></JumpLinksList>
+                      </JumpLinksItem>
+                    ))}
+                  </JumpLinks>
+                </SidebarPanel>
+                <SidebarContent>
+                  <Button onClick={onToggleClick}>Toggle drawer</Button>
+                  <TextContent>
+                    {headings.map(heading => (
+                      <div key={heading}>
```

Comment:
We should probably match the `div` styling here to the `div` styling in the other Jump Links demo! 

Looks like there's currently some missing spacing between the text content and the headers. Setting the `maxWidth` here keeps all the text visible when the drawer is toggled too 🙂

```suggestion
                      <div key={heading} style={{ maxWidth: '800px', marginBottom: '32px' }}>
```